fix(kselect): kselect reskin [KHCP-8998]

portikM opened the pull request:
# Summary

[Jira](https://konghq.atlassian.net/browse/KHCP-8998)
[Figma](https://www.figma.com/file/Yze0SWXl5nKjR0rFdilljK/Components?type=design&node-id=283%3A2898&mode=dev)

`KSelect` component reskin.

Links:

- [Kongponents sandbox](https://deploy-preview-1832--kongponents-sandbox.netlify.app/#/select)
- [Kongponents docs preview](https://deploy-preview-1832--kongponents.netlify.app/components/select.html)

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

adamdehaven reviewed (changes requested):
@Leopoldthecoder or @sumimakito it looks like the `autosuggest` prop is being removed in this PR.

Wasn't it added for Kong Manager specifically?

portikM commented:
> @Leopoldthecoder or @sumimakito it looks like the `autosuggest` prop is being removed in this PR.
> 
> Wasn't it added for Kong Manager specifically?

Small correction: the async `items` fetching functionality - which in my understanding is what we needed the `autosuggest` prop for is still supported with `enableFiltering` and `@query-change` event: https://deploy-preview-1832--kongponents.netlify.app/components/select.html#loading

sumimakito commented:
I tested the preview package with the public-ui-components repo. It worked normally with Kong Manager's Admin API filtering mechanism. IMO, it should be okay to remove this prop as long as we update our usages accordingly in the public-ui-components repo.

As @portikM pointed out, the filtering is implemented on the host side (public-ui-components in this case), and it appears that `autosuggest` was only introduced as a hint that actually controls the underlying filter things (e.g., `filterIsEnabled` previously).

One more minor thing: I noticed the behavior of `filteredItems` will be inverted after these changes: Previously, when `autosuggest` was present, we would not call the filter function, while now we will call a default filter function if we `enable-filtering` but not providing a `filter-function`. 

```ts
const filteredItems = computed(() => {
  return props.enableFiltering ? props.filterFunction({ query: filterQuery.value, items: selectItems.value }) : selectItems.value
})
```

https://github.com/Kong/kongponents/pull/1832/files#diff-68c30a7348bc67fac2678056261caf1e95de49eac77fe252f3efa4c7fe2eecf9R428-R430

Would it be better if we bypass the `props.filterFunction` call when it is `null` or non-function?

portikM commented:
> I tested the preview package with the public-ui-components repo. It worked normally with Kong Manager's Admin API filtering mechanism. IMO, it should be okay to remove this prop as long as we update our usages accordingly in the public-ui-components repo.
> 
> As @portikM pointed out, the filtering is implemented on the host side (public-ui-components in this case), and it appears that `autosuggest` was only introduced as a hint that actually controls the underlying filter things (e.g., `filterIsEnabled` previously).
> 
> One more minor thing: I noticed the behavior of `filteredItems` will be inverted after these changes: Previously, when `autosuggest` was present, we would not call the filter function, while now we will call a default filter function if we `enable-filtering` but not providing a `filter-function`.
> 
> ```ts
> const filteredItems = computed(() => {
>   return props.enableFiltering ? props.filterFunction({ query: filterQuery.value, items: selectItems.value }) : selectItems.value
> })
> ```
> 
> https://github.com/Kong/kongponents/pull/1832/files#diff-68c30a7348bc67fac2678056261caf1e95de49eac77fe252f3efa4c7fe2eecf9R428-R430
> 
> Would it be better if we bypass the `props.filterFunction` call when it is `null` or non-function?

Good call, thanks for testing this Makito! I added an option to bypass default filter function by passing a function that returns `true` through `filterFunction` prop: https://github.com/Kong/kongponents/pull/1832/commits/fe76192430bb235be9afaef9153ee52f64890d0e.

adamdehaven commented:
![image](https://github.com/Kong/kongponents/assets/2229946/1124f984-03c2-4f70-9dc5-886aa5219431)

A couple issues showing up on the sandbox:

1. The KSelect (and the sandbox container) extends beyond the right side of the page
2. The KSelect dropdown is wider than the input element

adamdehaven commented:
![image](https://github.com/Kong/kongponents/assets/2229946/ed4a2b18-8d8f-4b24-a7c6-86de870085f1)

on the enableItemCreation example, there is a bunch of extra space in the dropdown below the added item

adamdehaven commented:
![image](https://github.com/Kong/kongponents/assets/2229946/1b7dd8ea-8864-4cfe-9528-c52069ab2122)
In the sandbox, there's raw code - something isn't right

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
+### error
 
-You can configure the button text when an item is selected, if `appearance` is type `button`.
+When set to `true`, error styling will apply.
```
The input element has an `errorMessage` prop, but this component reuses the `help` prop? Seems confusing - do we have a standard?

portikM replied:
Added a quick and easy way to sanitize user input: https://github.com/Kong/kongponents/pull/1832/commits/42cd593c12afc33cf797146d3074c346daf35c59. Though I would argue that this would need to be a separate ticket.

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
+### enableItemCreation
 
-Use this prop to customize selected item element appearance by reusing content passed through [item-template](#item-template) slot. **Note:** this prop only applies when `appearance` prop is `select` (use [selected-item-template](#selected-item-template) slot in other cases).
+When used in conjunction with `enableFiltering` set to `true`, KSelect will suggest adding a new value when filtering produces no results.
```
issue: we need to sanitize HTML and input to prevent things like this (same comment on the multiselect PR)
![image](https://github.com/Kong/kongponents/assets/2229946/2e285420-1929-4eb5-a712-5dda5aeb49b0)

adamdehaven replied:
If a user enters HTML or script tags when creating a new item, it's adding that HTML directly into the DOM (XSS).

You need to sanitize the user input to prevent this.

portikM replied:
Added a quick and easy way to sanitize user input: https://github.com/Kong/kongponents/pull/1832/commits/42cd593c12afc33cf797146d3074c346daf35c59. Though I would argue that this would need to be a separate ticket.

adamdehaven replied:
This looks good; you'll need this on KMultiselect too

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
 ### dropdownFooterText
 
-Adds informational text to the bottom of the dropdown options which remains visible even if the content is scrolled. Can also be [slotted](#slots).
+Text to be displayed at the bottom of dropdown container. Can also be [slotted](#dropdown-footer-text).
```
```suggestion
Text to be displayed at the bottom of the dropdown container. Can also be [slotted](#dropdown-footer-text).
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
-If you want to override the behaviour and have the footer text at the end of the dropdown list, use the value `static`. This ensures the footer text is visible only when the user scrolls to view the bottom of the list.
-
-Accepted values: `sticky` (default) and `static`.
+Defaults to `sticky`, but also accepts `static` value should you want text passed through `dropdownFooterText` prop to bi displayed at the bottom of dropdown container after all items.
```
```suggestion
Defaults to `sticky`, but also accepts `static` value should you want text passed through `dropdownFooterText` prop to be displayed at the bottom of dropdown container after all items.
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
-Use this prop to control whether or not the `KSelect` component with an `appearance` prop set to a value of `select` or `dropdown` allows filtering. By default, filtering is enabled for `dropdown` appearance and is disabled for `select` appearance.
-
-`button` style `appearance` does not have filter support because it is a button.
+By default, items passed to KSelect are not searchable. When `enableFiltering` prop is true, clicking on select element will turn it into input field, typing into which will filter matching items.
```
```suggestion
By default, items passed to KSelect are not searchable. When `enableFiltering` prop is true, clicking on the select element will turn it into input field. You may filter items by typing in the input which will hide non-matching items.
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
-### v-model
-
-`KSelect` works as regular inputs do using v-model for data binding:
+Custom function to filter items by.
```
```suggestion
A custom function to perform item filtering.
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
-  </KSelect>
-</ClientOnly>
+:::tip TIP
+Should you need to disable the default filter function while still allowing filtering (for example when you want to handle filtering asynchronously by fetching from the API), you can pass a function that always returns boolean `true` to this prop, like so: 
```
```suggestion
Should you need to disable the default filter function while still allowing filtering (for example when you want to handle filtering asynchronously by querying an API), you can provide a function that always returns boolean `true` to this prop, like so: 
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
 
+### reuseItemTemplate
+
+Whether content passed through [`item-template` slot](#itemtemplate) should be applied for selected item.
```
```suggestion
Reuse the same display format provided via the [`item-template` slot](#itemtemplate) for the selected item.
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
-When `autosuggest` is enabled, you can use the `loading` prop to show a loading indicator while fetching data from API.
-By default, the loading indicator is a spinner icon, and you can implement your own indicator using the `loading` slot.
-See [autosuggest](#autosuggest) for an example.
+Use fixed positioning of the popover to avoid content being clipped by parental boundaries. See [KPopover](/components/popover#positionfixed) docs for more information.
```
```suggestion
Use fixed positioning for the select dropdown to avoid content being clipped by the boundaries of its parent container. See [KPopover](/components/popover#positionfixed) docs for more information.
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
+### loading
 
-Deselecting the item will completely remove it from the list and underlying data, and trigger a `item:removed` event containing the removed item's data.
+Pass `true` to display loader instead of items in the dropdown. KSelect's `item` prop is reactive to changes by design, so that should you need to perform async item fetching/filtering you can execute that logic on host app's side and pass items back to KSelect. The example below utilizes the [`@query-change` event](#query-change) to simulate async items fetching behind the scenes.
```
```suggestion
Pass `true` to display loader instead of items in the dropdown. KSelect's `item` prop is reactive to changes by design, so that should you need to perform async item fetching/filtering you can execute that logic within the host app and pass items back to KSelect. 

The example below utilizes the [`@query-change` event](#query-change) to simulate async item fetching behind the scenes.
```

adamdehaven commented on docs/components/select.md:
```diff
@@ -1,1013 +1,741 @@
-:::
+## Slots
+
+### label-tooltip
```
suggestion: Can we move `label-tooltip` below the other, more important slots?

adamdehaven commented on docs/guide/migrating-to-version-9.md:
```diff
@@ -316,6 +316,7 @@ This component has been removed. Please refer to KBadge component which has been
 
 ### KPagination
 
+KPagination now uses [KDropdown](/components/dropdown) instead of [KSelect](/components/select) as underlying element for page size selector.
```
```suggestion
KPagination now uses [KDropdown](/components/dropdown) instead of [KSelect](/components/select) as the underlying component for page size selection.
```

adamdehaven commented on docs/guide/migrating-to-version-9.md:
```diff
@@ -365,6 +366,37 @@ Removed as of `v9`. Use `KBreadcumbs` instead.
+
+#### Events
+
+* `item:added` event has been changed to `item-added`
+* `item:removed` event has been changed to `item-removed`
```
same question here as multiselect - why are we renaming the events?

adamdehaven commented on src/components/KInput/KInput.vue:
```diff
@@ -260,6 +263,21 @@ watch(() => props.error, (newVal, oldVal) => {
   }
 })
+
+const beforeSlotElement = ref<HTMLElement>()
+const afterSlotElement = ref<HTMLElement>()
```
nit: should be initialized as `null`
```suggestion
const beforeSlotElement = ref<HTMLElement | null>(null)
const afterSlotElement = ref<HTMLElement | null>(null)
```

adamdehaven commented on src/components/KPagination/KPagination.vue:
```diff
@@ -114,17 +114,21 @@
-          @selected="updatePageSize"
-        />
+          selection-menu
+          show-caret
```
<img width="742" alt="Screenshot 2024-01-03 at 9 46 47 PM" src="https://github.com/Kong/kongponents/assets/2229946/ceb31c8f-0e0f-435b-aa57-4369595f283d">
question: should we add a max-height attribute to this pagination dropdown so that it's not super-tall and just scrolls internally?

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -2,278 +2,195 @@
+        }"
+        :position-fixed="positionFixed"
+        target=".select-wrapper"
+        @closed="() => {
+          if (selectedItem) {
+            filterQuery = selectedItem.label
+          }
+          if (isToggled.value) {
```
suggestion: would it be cleaner to refactor these in-template function blocks `@closed` `on-popover-click` etc. to functions down in the script block? You should be able to pass around the diff callbacks/vars as needed; just wondering if it would be cleaner.

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -2,278 +2,195 @@
-              :style="widthStyle"
-              @keyup="(evt: any) => triggerFocus(evt, isToggled)"
+            <template #after>
+              <CloseIcon
+                v-if="isClearVisible"
+                data-testid="clear-selection-icon"
+                role="button"
+                tabindex="0"
+                @click="clearSelection"
+              />
```
issue: this clear icon needs a `focus-visible` rule for keyboard nav (box-shadow, etc.)

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -2,278 +2,195 @@
+          <div
+            v-if="enableFiltering && loading"
+            class="select-loading"
+            data-propagate-clicks="false"
```
question: what is `data-propagate-clicks`?

portikM replied:
Sometimes clicks on containers are propagated up which results in unintentional clicks on other elements. I added this data property to control that when we can't use `pointer-events: none;`.

adamdehaven replied:
can you document this in the comments somewhere?

portikM replied:
Added a comment in `onSelectWrapperClick` method.

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -366,25 +256,18 @@ const props = defineProps({
-    type: Boolean,
-    default: false,
+  filterFunction: {
+    type: Function,
```
question: is it possible to type this function any more specifically?

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -555,63 +418,43 @@ const popoverContentMaxHeight = computed((): string => getSizeFromString(props.d
-  }
-  return placeholderText.value
-})
+const placeholderText = computed((): string => props.placeholder || attrs.placeholder as string || 'Filter...')
```
Should the default placeholder be `Select` instead? If filtering isn't enabled it doesn't make much sense

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -555,63 +418,43 @@ const popoverContentMaxHeight = computed((): string => getSizeFromString(props.d
+const hasCustomSelectedItem = computed((): boolean => !!(selectedItem.value &&
   (slots['selected-item-template'] || (props.reuseItemTemplate && slots['item-template']))))
 
+const filteredItems = computed(() => {
+  return props.enableFiltering && props.filterFunction({ query: filterQuery.value, items: selectItems.value }) !== true ? props.filterFunction({ query: filterQuery.value, items: selectItems.value }) : selectItems.value
+})
```
This block is pretty busy - can you add comments or break it up at all?

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -671,34 +512,38 @@ const triggerFocus = (evt: any, isToggled: Ref<boolean>):void => {
 }
 
+const onSelectWrapperClick = (event: Event): void => {
+  if (isDisabled.value || (event?.target as HTMLElement)?.dataset.propagateClicks === 'false') {
```
nit: whole PR 

There are a ton of changes; can you add comments in the code please?

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -741,385 +586,162 @@ watch(() => props.items, (newValue, oldValue) => {
-
-.k-select {
-  width: fit-content; // necessary for correct placement of popup
+watch(filterQuery, () => {
+  emit('query-change', filterQuery.value)
+})
```
nit: use the watch value instead of the ref
```suggestion
watch(filterQuery, (q: string) => {
  emit('query-change', q)
})
```

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -741,385 +586,162 @@ watch(() => props.items, (newValue, oldValue) => {
+onMounted(() => {
+  if (selectWrapperElement.value) {
+    resizeObserver.value = new ResizeObserver(entries => {
+      // TODO: use resizeObserverHelper(entries, () => { actualElementWidth.value = `${selectWrapperElement.value?.offsetWidth}px` })
+      // Wrapper 'window.requestAnimationFrame' is needed for disabling "ResizeObserver loop limit exceeded" error in DD
```
question: is this TODO still valid?

adamdehaven commented on src/components/KSelect/KSelect.vue:
```diff
@@ -741,385 +586,162 @@ watch(() => props.items, (newValue, oldValue) => {
-    margin-left: var(--kui-space-40, $kui-space-40);
-    vertical-align: middle;
-    width: 0;
+onBeforeUnmount(() => {
```
question: `onUnmounted` instead?

kaiarrowood reviewed (changes requested):
Most of the functionality is good, but I noticed some things that don't appear right or are confusing.

Item creation appears to be broken, nothing happens when I hit enter - you will need to hook up to the emitted events and add the item into the list in order to verify the actual use case works as expected (and show devs how they are expected to use it).

![image](https://github.com/Kong/kongponents/assets/67973710/f2f66e84-9a88-4671-955c-3cce20170b79)

Too much bottom padding on empty state/item create?

![image](https://github.com/Kong/kongponents/assets/67973710/20bef0ef-0641-4196-9f10-128a4eeac52a)

![image](https://github.com/Kong/kongponents/assets/67973710/10e4fbca-e010-4ab5-8fe0-c146adba77cf)


Disabled state should have disabled cursor, not pointer:

![image](https://github.com/Kong/kongponents/assets/67973710/0dc4a7e0-9fd7-4618-8c0e-6a5244b36e4d)

Sandbox issue - same KSelectItem instance being used in 2 different KSelect's, clicking one triggers a click on the other

![image](https://github.com/Kong/kongponents/assets/67973710/5ded174b-795e-4c95-8d9d-7d60a3048909)

Two items selected? Possibly caused by reusing the KSelectItem?

![image](https://github.com/Kong/kongponents/assets/67973710/f69975f6-d74a-4e30-8d07-e85a07f32fa0)

![image](https://github.com/Kong/kongponents/assets/67973710/cdd1b692-709c-4902-8207-fe0fc70d02ae)

To actually test the functionality, this KSelect needs filtering enabled (input event should fire when you type): 

![image](https://github.com/Kong/kongponents/assets/67973710/c5c462d6-4965-4f72-97e0-08f4a478abf2)

Horizontal scrollbars in docs site example for `dropdownMaxHeight`:

![image](https://github.com/Kong/kongponents/assets/67973710/9c61d1d6-0158-4bb1-bb71-1b2573c5ecbb)

I'm not sure this is the best example, I sort of expect results for this query. Maybe make the other searchable field be a string instead of an array, or adjust your function to check if each item of the array includes the substring rather than just if the array includes it. I also recommend updating the item template to show the other field being searched so it's less visually confusing what's happening.

![image](https://github.com/Kong/kongponents/assets/67973710/1746dc34-aeb1-4dec-b8c3-4c85bee0fce2)

I understand we want to keep things simple but these examples feel broken since you can't see/use the non-loading/non-empty state. I'd kind of like to see something even as simple as a toggle or button to switch between normal state and whatever slot you're trying to show to ensure default functionality is working as expected with the slot.

![image](https://github.com/Kong/kongponents/assets/67973710/89bd709c-cf73-43d5-b788-f2565a10364e)

It would be really helpful if we can link to or display what we expect `item` to look like here

![image](https://github.com/Kong/kongponents/assets/67973710/5af7440f-f2d7-4bd4-bf71-f5f0cb8638a3)

portikM commented:
> Most of the functionality is good, but I noticed some things that don't appear right or are confusing.
> 
> Item creation appears to be broken, nothing happens when I hit enter - you will need to hook up to the emitted events and add the item into the list in order to verify the actual use case works as expected (and show devs how they are expected to use it).
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294219392-f2f66e84-9a88-4671-955c-3cce20170b79.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTkzOTItZjJmNjZlODQtOWE4OC00NjcxLTk1NWMtM2NjZTIwMTcwYjc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlM2MzNWQ0MDc1OTM1NmIxZmFkYjM1N2Y2ZDQ3YTU2OWFhZjU0MDc4NDMwZDYzYTUzMmZlZDhlNDA0OTNhOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qGnUOa4cy29wfXomMdFyyP32zK3u86NJIFNIRaP6dI0)
> 
> Too much bottom padding on empty state/item create?
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294214375-20bef0ef-0641-4196-9f10-128a4eeac52a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTQzNzUtMjBiZWYwZWYtMDY0MS00MTk2LTlmMTAtMTI4YTRlZWFjNTJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmMWMwZGRkNmZjMDA5NjBiODdiZTAwZTNkODAwYWMxMDI2ZDhkZDRjMTY2OTVlNmJmMDE5YWM0NzNkODc0YTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.u3GOt4Nbptw2N6uJokvj_L-8oWv-VAD4H-qGEDq-MmU)
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294215245-10e4fbca-e010-4ab5-8fe0-c146adba77cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTUyNDUtMTBlNGZiY2EtZTAxMC00YWI1LThmZTAtYzE0NmFkYmE3N2NmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwNzVlYWFlZGIyMTM1MmZlODBmNTIzNmM0YTBmNzk5ZDk1YmZjZTNjODU2MDRlZTg5YzE2NmE3YzRlMDYzN2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fFwMs0oF2KTb9DRKsQu2Lfj4XWfXZ63XKlnDq5D2y_k)
> 
> Disabled state should have disabled cursor, not pointer:
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294215027-0dc4a7e0-9fd7-4618-8c0e-6a5244b36e4d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTUwMjctMGRjNGE3ZTAtOWZkNy00NjE4LThjMGUtNmE1MjQ0YjM2ZTRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2OWRlMzMwZjc4YTI3YzZjMzMzNTQwNmZiYzcyOWEzMTA1NDk2NTM2YTZhY2QzMDUyYjE2NDg5ZTA5MjM2NzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aJj3ynCpWtqCn8J3KoJb06jiwAj6TilFpX6FBV7dNt4)
> 
> Sandbox issue - same KSelectItem instance being used in 2 different KSelect's, clicking one triggers a click on the other
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294215725-5ded174b-795e-4c95-8d9d-7d60a3048909.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTU3MjUtNWRlZDE3NGItNzk1ZS00Yzk1LThkOWQtN2Q2MGEzMDQ4OTA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkYjU1YTZjMGE5MmJmNTJjNjM2YmIxZGM4ZjI3Nzg1OWQ2NTIyNjUxYTM0MmRiYjE0ZWNhZTc0ODI1NTFmYmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2S0pRasUsvoSzYwmu6QPnxL8Lr0-ngN9jcMCOkzmtdk)
> 
> Two items selected? Possibly caused by reusing the KSelectItem?
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294216053-f69975f6-d74a-4e30-8d07-e85a07f32fa0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTYwNTMtZjY5OTc1ZjYtZDc0YS00ZTMwLThkMDctZTg1YTA3ZjMyZmEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjYmQ4YmJlMzZkZWQ1MDk5YThiZTEwYWVmYTdhODk5YTI1Y2FhZTZhYWVkNjdjZWJhMjcxMmJkZmUzY2Y5NWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.p5A434a9Era2nBEFYPDy_tNq4NesUgiITgxaXNZ9Yos)
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294216148-cdd1b692-709c-4902-8207-fe0fc70d02ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTYxNDgtY2RkMWI2OTItNzA5Yy00OTAyLTgyMDctZmUwZmM3MGQwMmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiZWU4NjdjZWQwYzI0MjU0MDYzNmJjMGU1ZDE5M2U0MWM4M2M3N2ZiYmQzZmJlM2E1YzM0M2VhN2Q2NjdiYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DgjGLILDXInM0KGcyviNRxg5K8WQfMqfmxtEveyFz0Y)
> 
> To actually test the functionality, this KSelect needs filtering enabled (input event should fire when you type):
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294216408-c5c462d6-4965-4f72-97e0-08f4a478abf2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTY0MDgtYzVjNDYyZDYtNDk2NS00ZjcyLTk3ZTAtMDhmNGE0NzhhYmYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhM2VmZDQyMzM5ODliMzZiMmE4YzIwMWUwOTFlZmFiYTVkOWNjYjViOWEwZGI3YTNiYWZiNDc0M2NiMmMwMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kSpjoU12ps-sz12e00wJTOa6HHB-cPOY87woG7dk4Q4)
> 
> I'm not sure this is the best example, I sort of expect results for this query. Maybe make the other searchable field be a string instead of an array, or adjust your function to check if each item of the array includes the substring rather than just if the array includes it. I also recommend updating the item template to show the other field being searched so it's less visually confusing what's happening.
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294219834-1746dc34-aeb1-4dec-b8c3-4c85bee0fce2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTk4MzQtMTc0NmRjMzQtYWViMS00ZGVjLWI4YzMtNGM4NWJlZTBmY2UyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2NmZmMGU4MWJhY2ViNjJlMjA3MmQyNjI2OGE5N2U4ZTZiZmI5ZmI0YTc4ZDU1NDZmMTVlNzE3Mzg5MjE1NmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8xnml3lO0ZTN78TQTmS6HbCFovrbaBkFY6Fpx0ajLy0)
> 
> I understand we want to keep things simple but these examples feel broken since you can't see/use the non-loading/non-empty state. I'd kind of like to see something even as simple as a toggle or button to switch between normal state and whatever slot you're trying to show to ensure default functionality is working as expected with the slot.
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294221487-89bd709c-cf73-43d5-b788-f2565a10364e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMjE0ODctODliZDcwOWMtY2Y3My00M2Q1LWI3ODgtZjI1NjVhMTAzNjRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlOTVhN2MyYTRhNDgzMjgxNDk1ODViZThmMjJiNTk5YTA4YzNmY2MwZDRlZDNlNDkyMDhjNDQ2MGU3NjI1MjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AtTgp-NkoLRx0SKDcv-EQ6yCk0cIQUGk8fgAnjNnoS0)
> 
> It would be really helpful if we can link to or display what we expect `item` to look like here
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294219694-5af7440f-f2d7-4bd4-bf71-f5f0cb8638a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDYxNzcsIm5iZiI6MTcwNDQwNTg3NywicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTk2OTQtNWFmNzQ0MGYtZjJkNy00YmQ0LWJmNzEtZjVmMGNiODYzOGEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyMDQzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwZjA0NWUwODNhY2Y4YTM5MjlkNjI0ZGQ1ZTllOGRiMTM0NDUzOGJjZmFiYzQ2NjlkOTI4YjBkZmIxY2NiYzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0XIoysQseIRgWbJq2MprtABQ6P8ZdQ7g3yB1Epf4t6Y)

Thanks for catching bugs here! Fixed a few bugs outlined here and added some improvements in documentation like specifying interfaces and better examples.

portikM commented on src/components/KSelect/KSelect.vue:
```diff
@@ -671,34 +506,68 @@ const triggerFocus = (evt: any, isToggled: Ref<boolean>):void => {
 }
 
+const onSelectWrapperClick = (event: Event): void => {
+  console.log('here')
```
```suggestion
```

adamdehaven commented:
issue: I can tab to the "clearable" button, but I can't press any key to actually clear the select
![image](https://github.com/Kong/kongponents/assets/2229946/e5372805-edf0-4e29-821a-c5eec2a99e1a)

kaiarrowood commented:
- bug: Item creation has a bug when hitting enter - my typed text is cleared. We should do nothing instead.

- bug: still seeing this
> Two items selected? Possibly caused by reusing the KSelectItem?
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294216053-f69975f6-d74a-4e30-8d07-e85a07f32fa0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzI5NDYsIm5iZiI6MTcwNDczMjY0NiwicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTYwNTMtZjY5OTc1ZjYtZDc0YS00ZTMwLThkMDctZTg1YTA3ZjMyZmEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDE2NTA0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4MWY2NDA5M2JlYjUzMTVmYTI2MmY1MzViNjE3ZmEwYWEyOGFjYTQ4Yzk2ODA2ODU4NzdkYzJlMzI1NzJkYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cPsF3hcpht10d_u01y2xguluxAGcBsBtWuA6LhGmvXQ)
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294216148-cdd1b692-709c-4902-8207-fe0fc70d02ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzI5NDYsIm5iZiI6MTcwNDczMjY0NiwicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTYxNDgtY2RkMWI2OTItNzA5Yy00OTAyLTgyMDctZmUwZmM3MGQwMmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDE2NTA0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3ODExODAyMGQzNDRmM2UyMTIzMTNjZDc3ODY4ZWI4MWU5OTc0NDcyNTI1NGE0NWU5OWI3ODk0YjgyM2EwMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SzImGemxgZjRf2311zScHyoBLj2B9fhZ4VKP0S7xU1Q)

- request: Please can we do this?
> I'm not sure this is the best example, I sort of expect results for this query. Maybe make the other searchable field be a string instead of an array, or adjust your function to check if each item of the array includes the substring rather than just if the array includes it. **I also recommend updating the item template to show the other field being searched so it's less visually confusing what's happening.**
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294219834-1746dc34-aeb1-4dec-b8c3-4c85bee0fce2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzI5NDYsIm5iZiI6MTcwNDczMjY0NiwicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTk4MzQtMTc0NmRjMzQtYWViMS00ZGVjLWI4YzMtNGM4NWJlZTBmY2UyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDE2NTA0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkYTVhY2ViMTZiN2UyZTE0ZjM3ZTI4MmIyYTcxMGNhZmU3YTQ4YTQ0MjA0NTVmYzlhZDJiZDU5ZjdkMjk0NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jznVSIqE8MV8hqDznuPWBzSCZWfcMsTQ4JGLqDfKb6s)

- suggestion: I still think this is a problem. We have basically provided non-usable examples (ex. a user is never going to have a scenario where you want loading state to always be visible). We could easily improve these examples by having a toggle for loading and whether or not items is an empty array.
> I understand we want to keep things simple but these examples feel broken since you can't see/use the non-loading/non-empty state. I'd kind of like to see something even as simple as a toggle or button to switch between normal state and whatever slot you're trying to show to ensure default functionality is working as expected with the slot.
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294221487-89bd709c-cf73-43d5-b788-f2565a10364e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzI5NDYsIm5iZiI6MTcwNDczMjY0NiwicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMjE0ODctODliZDcwOWMtY2Y3My00M2Q1LWI3ODgtZjI1NjVhMTAzNjRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDE2NTA0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjYTA4YzVhYzc4NTMwYzZjMWM4MDI2YTEwMzZmYjhiODMxZTMzYzBiY2RhYmYwZmU2YjdmYTYwODEzYmJhYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cP0nrfCn7NUH5GRpyiW0oG-R5_XUgYwDmsQv4PZRh9U)

- request: Please can we do this? You can just link to [here](https://deploy-preview-1832--kongponents.netlify.app/components/select.html#items) since the first thing under that prop is the interface for `SelectItem`.
> It would be really helpful if we can link to or display what we expect `item` to look like here
> 
> ![image](https://private-user-images.githubusercontent.com/67973710/294219694-5af7440f-f2d7-4bd4-bf71-f5f0cb8638a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MzI5NDYsIm5iZiI6MTcwNDczMjY0NiwicGF0aCI6Ii82Nzk3MzcxMC8yOTQyMTk2OTQtNWFmNzQ0MGYtZjJkNy00YmQ0LWJmNzEtZjVmMGNiODYzOGEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDE2NTA0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjMGRiN2NjMjRjOWViYjJjZTY5ODAyNzBkNWVlZjgxMzEzMGRiOWM2ZTgyYzU5MGU0ZGQ4ZThlYjBiMjBmZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CS3okROh85BS8emq925liXMPRsSJuaaShlgv9kUkxj0)

portikM commented:
> ![image](https://private-user-images.githubusercontent.com/2229946/294071778-1b7dd8ea-8864-4cfe-9528-c52069ab2122.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3NDIxNzYsIm5iZiI6MTcwNDc0MTg3NiwicGF0aCI6Ii8yMjI5OTQ2LzI5NDA3MTc3OC0xYjdkZDhlYS04ODY0LTRjZmUtOTUyOC1jNTIwNjlhYjIxMjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDhUMTkyNDM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDMyMTliMjhkODVjNGZiMGEyNTE4ZTVjMTNmNzExMGU2MjE3MGFhNmE1MTExZjNmNmU2ODM4MzJkZmMxNjhiNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.P7UZU6ydd4yKS_BamVMQAc6YN_UqiLMjy4R4YxhDujM) In the sandbox, there's raw code - something isn't right

It's an example

kaiarrowood reviewed (commented):
Functionally verified sandbox and docs working as expected 🚀
